extensions/mapreduce/src/main/java/org/radargun/stages/mapreduce/MapReduceStage.java

```diff
@@ -121,20 +126,31 @@
    @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
    private MapReducer<KOut, VOut, R> mapReducer;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait(dependency = InjectTrait.Dependency.OPTIONAL)
    private Clustered clustered;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait
    private CacheInformation cacheInformation;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait
    private BasicOperations basicOperations;
 
    @InjectTrait(dependency = InjectTrait.Dependency.OPTIONAL)
    private Iterable iterable;
 
    private boolean supportsResultCacheName = false;
 
+   @Init
+   public void validate() {
```

Comment:
The implementation will naturally depend on the existence of a cache, be it remote or embedded one. But the MR stage shouldn't require cache-related traits. The source of the data will be just and identifier (e.g. cache name), and the implementation of the trait is free to interpret it.

The whole point of Traits is to cover vendor-specific API (in as lightweight wrapper as possible), and let benchmark code operate only on this wrapped one.

To put both into one benchmark, I would be more troubled about the setup, since Infinispan would have some groups just mocked, and when X is in Spark executed on slave A and Y on slave B, in Infinispan both should go to single node.